src/scenes/Dashboard/ProgressReportsWidget/ProgressReportsExpandedGrid.tsx

```diff
@@ -128,6 +138,21 @@ export const ProgressReportsExpandedGrid = (
     [onMouseDown]
   );
 
+  const onStateChange = useDebounceFn(
+    () => {
+      const gridState = apiRef.current.exportState();
+      if (!isEqual(gridState, prevState.current)) {
+        prevState.current = gridState;
+        setSavedGridState(gridState);
+      }
+    },
+    { wait: 500, maxWait: 500 }
+  );
+
+  useEffect(() => {
+    apiRef.current.restoreState(savedGridState);
+  }, [savedGridState, apiRef]);
```

Comment:
Let's create a hook to wrap these ~4 hooks, so we don't have to repeat ourselves for every grid.

Reply:
Agreed, that seems like a natural refactor.